### PR TITLE
Add Baremetal hosts to overview dashboard inventory

### PR DESCRIPTION
**Pull request by rawagner**
@jtomasek @honza I haven't tried it so if you can, it would be great

verified, thanks @honza 
bmh_inventory

### CODE REVIEW

**Comment by rawagner** [comment]
@honza @jtomasek take a look

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  HOST_REGISTERING_STATES,
+  HOST_SUCCESS_STATES,
+} from '../../constants';
+
+const BMH_STATUS_GROUP_MAPPER = {
+  [InventoryStatusGroup.OK]: [...HOST_SUCCESS_STATES],
+  [InventoryStatusGroup.PROGRESS]: [
+    ...HOST_PROGRESS_STATES,
+    ...HOST_PROVISIONING_STATES,
+    ...HOST_REGISTERING_STATES,
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: `HOST_REGISTERING_STATES` and `HOST_PROVISIONING_STATES` are not needed as those are a subset of `HOST_PROGRESS_STATES`

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  ],
+  [InventoryStatusGroup.ERROR]: [...HOST_ERROR_STATES],
+};
+
+export const getBMHStatusGroups: StatusGroupMapper = (bmhs) => {
+  const groups = {
+    [InventoryStatusGroup.OK]: {
+      statusIDs: ['ready', 'provisioned'],
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: I believe the statusIDs be the host status constants from `../../constants`

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    [InventoryStatusGroup.NOT_MAPPED]: {
+      statusIDs: ['other'],
+      count: 0,
+      filterType: 'host-status',
+    },
+  };
+
+  bmhs.forEach((bmh) => {
+    const status = getSimpleHostStatus(bmh);
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: This function does not exist any more. `getBMHStatusGroups` will now need to take `bmh` and `nodeMaintenance` to correctly identify host status. See https://github.com/openshift/console/blob/master/frontend/packages/metal3-plugin/src/utils/host-status.ts#L57

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts (hunk 1 above)
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: good to know, removed redundant states

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts (hunk 3 above)
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: function updated. Im fetching bmhs, nodes, hosts and node maintenances now

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts (hunk 2 above)
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: these statusIDs are keys in this object https://github.com/openshift/console/blob/master/frontend/packages/metal3-plugin/src/components/table-filters.ts#L14 .  Most of them are in constants (missing `other`) but It seems to me that those constants are values which can appear in bmh status K8s object and are not meant to be IDs of status filters. Correct me if Im wrong.

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      count: 0,
+      filterType: 'host-status',
+    },
+  };
+
+  bmhs.forEach((bmh) => {
+    const machine = getHostMachine(bmh, machines as MachineKind[]);
+    const node = getMachineNode(machine, nodes as NodeKind[]);
+    const nodeMaintenance = findNodeMaintenance(maintenances, getName(node));
+    const hostMultiStatus = getHostStatus({ host: bmh, nodeMaintenance });
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: Nit: renaming `bmh` to host would align to what we're using in other places for baremetal host variable and simplify
```suggestion
    const hostMultiStatus = getHostStatus({ host, nodeMaintenance });
```

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    const status = getHostFilterStatus({
+      ...bmh,
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: same nit as above

File: frontend/packages/metal3-plugin/src/components/dashboard/utils.ts (hunk 2 above)
Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/dashboard/utils.ts]: You're right, thanks for clarification.

**Comment by rawagner** [comment]
renamed `bmh` to `host` to align with rest of metal3 code.

**Comment by spadgett** [comment]
/hold

It looks something in this pr has broken the tests

**Comment by rawagner** [comment]
Dashboards page failed to load because there some circular dependency when loading code from `plugin.ts` which depends on some code from `components/utils`. In this case the issue appeared after merging https://github.com/openshift/console/pull/1697. `plugin.ts` depends on code which is loading `Status` from `console/shared` and that depends on code from `components/utils` so I had to change imports of `console/shared` to more specific ones ie `console/shared/src/selectors`.

This is not specific to `Status` component as I noticed similar issue when `plugin.ts `depends on code from `components/utils` a few times before. We will need to take a closer look at the issue in the future.

**Pull request merged**